packages/@glimmer/util/lib/debug-to-string.ts

```diff
@@ -0,0 +1,25 @@
+import { DEBUG } from '@glimmer/env';
+
+let debugToString: undefined | ((value: unknown) => string);
+
+if (DEBUG) {
+  debugToString = (value: unknown) => {
+    let string;
+
+    if (typeof value === 'function') {
+      string = value.name;
+    } else if (typeof value === 'object' && value !== null) {
+      let className = (value.constructor && value.constructor.name) || '(unknown class)';
+
+      string = `(an instance of ${className})`;
+    } else if (value === undefined) {
+      string = '(an unknown tag)';
+    } else {
+      string = String(value);
+    }
+
+    return string;
+  };
+}
```

Comment:
Should this just be a no-op function in prod? The module can't be removed anyway so I don't know if it's worth making every caller care about it. If it matters (probably not), we should just make it a macro.